src/plugins/data/common/search/aggs/buckets/create_filter/date_histogram.test.ts

```diff
@@ -63,7 +63,7 @@ describe('AggConfig Filters', () => {
         max: bucketStart.clone().add(timePad),
       });
       agg.buckets.setInterval(interval);
-      filter = createFilterDateHistogram(agg, bucketKey);
+      filter = createFilterDateHistogram(agg, bucketKey) as RangeFilter;
```

Comment:
Why do we need the cast here? Can we extend `RangeFilter` to work without the cast here?

Reply:
Since `createFilterDateHistogram` can also return `ScriptedRangeFilter`, and then bellow I would have to handle optional props.